### DISCUSSION TITLE
Is element chaining available in 7.x (async)

### DISCUSSION THREAD

**Discussion by alisman** [webdriverio/webdriverio#12230] [Q&A]
I am transitioning to async.  I upgraded wdio to 7 because i want to do gradual transition to async.   There is a [blog post](https://blog.kiprosh.com/webdriverio-elements-chaining/) that says that chaining is available in 7.9.    It doesn't seem like there is actually a 7.9 release.  7.35 would seem to be latest.  Is async chaining available (before v8)?

**Comment by christian-bromann** [comment]
Yes.

**Comment by alisman** [comment]
@christian-bromann first of all, thank you so much for your timely responses on this project.   incredible!  we are using WDIO for a cancer research application (www.cbioportal.org) so please know that your effort is going toward a greater good!

Chaining doesn't seem to be working for me in 7.35.  I have the sync module installed because most of my tests are still using sync.  The following code is run inside of an async test block  it("..", async ()=>....).    I still get ```waitForExist is not a function```.  It works if I first assign to a variable and THEN call it.  I'm really hoping chaining will work because I have a TON of tests that depend on chaining.

```
            await $('[data-test="mutation-rate-summary"]').waitForExist({
                timeout: 60000,
            });
```

![image](https://github.com/webdriverio/webdriverio/assets/186521/0c289ef6-f0f1-4f02-961a-5ac68e30067b)

**Reply by christian-bromann** [reply]
> I have the sync module installed because most of my tests are still using sync.

I recommend to migrate over as sync mode has been deprecated and doesn't work unless you are using Node v14. Also your example uses `await` which is async code and should be incompatible when mixed together with sync code.

**Comment by alisman** [comment]
@christian-bromann we are trying to migrate but it would help to be able to do it slowly and documentation says explicity we can do that, which implies that both can be used at the same time no?

**Reply by christian-bromann** [reply]
Yes, you should be able to run async and sync test in one test suite, however you can not have sync and async commands next to each other.
